scripts/create-page.js

```diff
@@ -0,0 +1,168 @@
+const markdownSequence = ['Overview', 'Dos', 'Donts', 'BestPractices', 'Usage', 'Design', 'Contact', 'Custom', 'Markdown', 'Related'];
+
+// Paths/File Names
+const rootPath = './apps/fabric-website/src/';
+const pagePath = rootPath + 'pages/' + newPagePath;
+const pageFolderPath = pagePath + '/' + newPageName + 'Page/';
+const pageNamePrefix = pageFolderPath + newPageName + 'Page';
+const pageDocsPathRoot = pageFolderPath + 'docs/';
+const pageDocsPath = pageDocsPathRoot + 'default/';
+
+let templateFolderPath = './scripts/templates/create-page';
+
+const outputFiles = {
+  Page: pageNamePrefix + '.ts',
+  PageBase: pageNamePrefix + '.base.tsx',
+  PageModule: pageNamePrefix + '.tsx',
+  Styles: pageNamePrefix + '.styles.ts',
+  Module: pageNamePrefix + '.module.scss',
+  Doc: pageNamePrefix + '.doc.ts',
+  Types: pageNamePrefix + '.types.ts'
+};
+
+const markdownFiles = {
+  Overview: pageDocsPath + newPageName + 'Overview.md',
+  Dos: pageDocsPath + newPageName + 'Dos.md',
+  Donts: pageDocsPath + newPageName + 'Donts.md',
+  BestPractices: pageDocsPath + newPageName + 'BestPractices.md',
+  Usage: pageDocsPath + newPageName + 'Usage.md',
+  Design: pageDocsPath + newPageName + 'Design.md',
+  Contact: pageDocsPath + newPageName + 'Contact.md',
+  Custom: pageDocsPath + newPageName + 'Custom.md',
+  Markdown: pageDocsPath + newPageName + 'Markdown.md',
+  Related: pageDocsPathRoot + newPageName + 'Related.md'
+};
+
+// Error strings
+const errorCreatingPageDir = 'Error creating page directory';
+const errorPageName = 'Please pass in the page name using --name';
+const errorPagePath = 'Please pass in the page path using --path. ie: Overviews, Controls, Styles, etc';
+const errorPagePathDoesNotExist = "The path you gave doesn't exist. Please pass in a path that exists in the pages directory.";
+
+const errorUnableToOpenTemplate = templateFile => `Unable to open mustache template ${templateFile} for page`;
+const errorUnableToWriteFile = step => `Unable to write ${step} file`;
+
+// Success strings
+const successPageCreated = 'New page ' + newPageName + ' successfully created. Creating markdown files...';
+const successMarkdownCreated = 'Markdown files successfully created.';
+
+function handleError(error, errorPrependMessage) {
+  if (error) {
+    console.error(errorPrependMessage, error);
+    return false;
+  } else {
+    return true;
+  }
+}
+
+function createPageFiles(sequence, stepIndex) {
+  if (stepIndex >= sequence.length) {
+    // Success! The page has been created.
+    console.log(successPageCreated);
+    fs.mkdirSync(pageDocsPathRoot);
+    fs.mkdirSync(pageDocsPath);
+    createMarkdownFiles(markdownSequence, 0);
+    return;
+  }
+  const step = sequence[stepIndex];
+  const mustacheTemplateName = `Empty${step}.mustache`;
+
+  console.log('Creating ' + outputFiles[step] + '...');
+
+  fs.readFile(templateFolderPath + '/' + mustacheTemplateName, 'utf8', (error, data) => {
```

Comment:
The syntax for this gets a lot simpler if you use `fs-extra`, which supports promises (and therefore async/await).

Reply:
Thanks for the info. This is a modified version of `create-component`, since I'm not very experienced creating node scripts yet.

Reply:
@ecraig12345 I converted it to fs-extra and simplified some of the functions with promises.